Fetch notifications

PavanTeja2005 opened the pull request:
Fix #19 
I have added the notification fetching function "get_notification"

If _from and _to are given the notifications are fetched.
If only _from is given, from that date notifications are fetched
If none is given, previous 30 days notification are fetched

PavanTeja2005 committed:
notification fetcher

cro2003 commented:
On my end its showing `[]`

PavanTeja2005 commented:
It's because there are no notifications in the last 30 days. I have checked the site.

You can try using _from and _to..

cro2003 commented:
I see, as these are archived notices. So what you can do is that if there's no date is passed in argument just give all the important alerts which is showed in the pop when we open rgpv homepage + archived notice if its within one month range. If dates are passed and there are no notifications just give all the important alerts which is showed in the pop when we open rgpv homepage, here remember there can be case like i passed 01-08-2024 & 02-8-2024 there are notification between them but after these dates archived notifications are available in this case it shoud return []. In a bit i will be sharing some examples which will show how it should work.

cro2003 commented:
Example 1: No Arguments
Show all the Notification which are in the homepage of rgpv + archived notification which are within 1 month time, for instance today is 08-10-2024, check for the notification which are after 08-09-2024 in the archived notification. 
<img width="769" alt="image" src="https://github.com/user-attachments/assets/c8a971ee-50f3-40f7-9c21-83ce752567b8">

cro2003 commented:
Example 2: Argument Passed (_from: 01-09-2024, _to: 10-09-2024)
Show Notification from 01-09-2024 in archived notification + All the notifications from the Homepage (as after 04-09-2024 there are no archived notification but new notifications can be there as the _to date is 10-09-2024)
<img width="947" alt="image" src="https://github.com/user-attachments/assets/7c6d675e-f5c6-4238-9aa9-680d2d68d40f">
<img width="769" alt="image" src="https://github.com/user-attachments/assets/c8a971ee-50f3-40f7-9c21-83ce752567b8">

cro2003 commented:
Example 3: Argument Passed (_from: 10-09-2024, _to: 01-10-2024)
There are no notification from 10-09-2024 in archived notification, the latest notification was on 04-09-2024. So now in this case we can directly show the notification from the homepage
<img width="769" alt="image" src="https://github.com/user-attachments/assets/c8a971ee-50f3-40f7-9c21-83ce752567b8">

PavanTeja2005 commented:
There is no date data available. but is this okay? 
![image](https://github.com/user-attachments/assets/316aaef8-da89-441f-883b-a43b14451bc2)

PavanTeja2005 commented:
Archived case is fine, but it doesn't have any date info.

cro2003 commented:
Yea its cool but just add a key of date and write the value as "NOT AVAILABLE", `"date": "NOT AVAILABLE"`

cro2003 commented:
Also please make sure all the notification have the links to it (even with archived notification) as without link they will be useless.

cro2003 commented:
Example 4: Argument Passed (_from: 01-09-2024, _to: 03-09-2024)
In this Case there are no notification in the archived between the given dates, also we dont have to give the all notification from the homepage as the latest archived notification is after the _to date.
Output: []

cro2003 commented:
Example 5: Argument Passed (_from: 01-08-2024, _to: 01-09-2024)
As we can see that there are archived notification between these days so we only have to show that notifications, we dont have to give the all notification from the homepage as the latest archived notification is after the _to date.
<img width="763" alt="image" src="https://github.com/user-attachments/assets/79948021-c47c-44e5-b7dc-7dadac7a8661">

cro2003 commented:
Also add this functionality in `info_api.py` instead of `result_api.py`

PavanTeja2005 committed:
changes

PavanTeja2005 commented:
Check the new codes

Output:
![image](https://github.com/user-attachments/assets/8690847f-efaf-49c1-a04b-4e15f9c6b384)

cro2003 commented:
Can you please share the Output of the all 5 Examples which i have given above

PavanTeja2005 commented:
Example 1: No Arguments
Show all the Notification which are in the homepage of rgpv + archived notification which are within 1 month time, for instance today is 08-10-2024, check for the notification which are after 08-09-2024 in the archived notification.

![image](https://github.com/user-attachments/assets/252e07c0-37d2-4df9-9e06-43c9d3cea1f3)

PavanTeja2005 commented:
Example 2: Argument Passed (_from: 01-09-2024, _to: 10-09-2024)
Show Notification from 01-09-2024 in archived notification + All the notifications from the Homepage (as after 04-09-2024 there are no archived notification but new notifications can be there as the _to date is 10-09-2024)

![image](https://github.com/user-attachments/assets/430077b9-cefa-4155-9831-3070639b6382)

PavanTeja2005 commented:
Example 3: Argument Passed (_from: 10-09-2024, _to: 01-10-2024)
There are no notification from 10-09-2024 in archived notification, the latest notification was on 04-09-2024. So now in this case we can directly show the notification from the homepage
Start
![image](https://github.com/user-attachments/assets/1d7c5e25-d47c-4f43-810a-44fa7ba880d2)
end
![image](https://github.com/user-attachments/assets/adf08ff6-40e3-4d44-b67b-69be46187003)

PavanTeja2005 commented:
Example 4: Argument Passed (_from: 01-09-2024, _to: 03-09-2024)
In this Case there are no notification in the archived between the given dates, also we dont have to give the all notification from the homepage as the latest archived notification is after the _to date.
Output: []

![image](https://github.com/user-attachments/assets/d4e62d0f-9f18-42e6-b39b-9e067a372714)

PavanTeja2005 commented:
Example 5: Argument Passed (_from: 01-08-2024, _to: 01-09-2024)
As we can see that there are archived notification between these days so we only have to show that notifications, we dont have to give the all notification from the homepage as the latest archived notification is after the _to date.
image

![image](https://github.com/user-attachments/assets/df8ad2fc-b180-44b2-b976-80a948a5815b)

PavanTeja2005 commented:
What should I do with the archived notifications? As it's using js we cannot use requests. Using selenium will increase the project's complexity.

PavanTeja2005 commented:
I meant the links in the archived notifications.

cro2003 commented:
I see, let me check what be done here.

PavanTeja2005 committed:
Final changes

PavanTeja2005 commented:
Did you check?

cro2003 commented:
Yea I checked, so just give this link https://www.rgpv.ac.in/Uni/ImpNoticeArchive.aspx for the archived notification

PavanTeja2005 committed:
Update result_api.py

Added final modification. archive notifications page url added to the output json for the same.

PavanTeja2005 commented:
The last change done. Please approve.

And please let me know if there are any important features to be added more in the project.

PavanTeja2005 commented:
PLease merge this

cro2003 commented:
Hey So Sorry, I am still busy. I will review and merge it within 48 hours.

PavanTeja2005 commented:
Thank you

cro2003 commented:
I have tested the Code from my Side too and it works well. Before merging it please Move this notification code to the info_api.py file instead of result_api.py file and also modify the __init__.py file and at last the Alert from the rgpv homepage title also shows "Click here to View" along with the title so remove that too.

PavanTeja2005 committed:
Changed from results to info

PavanTeja2005 commented:
Modified according to requirements

cro2003 commented:
`get_notification()` & `fetch_alerts()` are still available in the result_api.py and it would be great you can move both both `get_notification()` & `fetch_alerts()` in the class as the static method, right now they are both outside the class [info_api.py file].
